tests/unit/useViolationsTest.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable rulesdir/prefer-underscore-method -- Disabled because it incorrectly flags the use of `it.each` */
+import { renderHook } from '@testing-library/react-native';
+import { useViolations, violationFields } from '@libs/Violations/useViolations';
+
+
+// Derive test cases from the violationFields object
+const allViolations = Object.keys(violationFields).map((name) => ({name}));
+// A list of all the field names that appear in `violationFields`
+const fieldNames = [...new Set(Object.values(violationFields))];
+// create an index of violations by field name
+const violationsByField = Object.entries(violationFields).reduce(
+    (acc, [violation, field]) => ({
+        ...acc,
+        [field]: [...(acc[field] || []), {name: violation}],
+    }),
+    {},
+);
+
+describe('useViolations', () => {
+    let violations = [];
+
+    beforeEach(() => {
+        violations = allViolations;
+    });
+
+    // The happy path
+    function callHook() {
+        const {result, rerender} = renderHook(() => useViolations(violations));
+        return {result, rerender};
+    }
+
+    it('returns correct values when there is only a single violation', () => {
+        const violation = violations[0];
+        violations = [violation];
```

Comment:
```suggestion
```

Reply:
This is needed? Otherwise it won't be testing the described case. It'll be using the default value of violations which is an array of all possible violations.

Reply:
Ah, right! This is why I prefer having the values we're testing defined near the place we run the assertions